File: Web/src/api/ctfUser.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetCTFUserList(data) {
  return apiClient.post("/ctfUser/GetCTFUserList", data).then(handleResponse).catch(handleError)
}

function AddCTFUser(data) {
    return apiClient.post("/ctfUser/AddCTFUser", data).then(handleResponse).catch(handleError)
}

function RemoveCTFUser(data) {
    return apiClient.post("/ctfUser/RemoveCTFUser", data).then(handleResponse).catch(handleError)
}

const ctfUserApi = {
  GetCTFUserList,
  AddCTFUser,
  RemoveCTFUser
}

export default ctfUserApi

File: Server/route/ctfUser.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetCTFUserRoute(r *gin.Engine) {
	adminCTFUser := r.Group("/ctfUser")
	adminCTFUser.Use(middleware.AdminAuthMiddleware())
	{
		adminCTFUser.POST("/GetCTFUserList", api.GetCTFUserList)
		adminCTFUser.POST("/AddCTFUser", api.AddCTFUser)
		adminCTFUser.POST("/RemoveCTFUser", api.RemoveCTFUser)
	}
}
